### PR TITLE
Deprecate `gate_set` parameter on engine classes.

### LINKED ISSUE #4995: Deprecate gate_set args in various Engine methods
As a follow-up to #4983, there are several methods in `Engine` and related classes that take a `gate_set` arg that should be deprecated. See the comment by @verult in https://github.com/quantumlib/Cirq/pull/4983#discussion_r806220843.

### PR DESCRIPTION
**Pull request by maffoo**
Fixes #4995

### CODE REVIEW

File: cirq-google/cirq_google/calibration/workflow.py [hunk 1]
```diff
@@ -842,25 +842,21 @@ def run_calibrations(
...
-        gate_set = sampler._gate_set
     else:
         engine = None
 
     if engine is not None:
         if processor_id is None:
-            raise ValueError('processor_id must be provided.')
-        if gate_set is None:
-            raise ValueError('gate_set must be provided.')
+            raise ValueError('processor_id must be provided.')  # coverage: ignore
```
Comment (verult) [inline cirq-google/cirq_google/calibration/workflow.py:850]: Did this file exist before coverage test was in place? Why does coverage check fail now but not before?

**Review by verult — COMMENTED** [review]
Thanks for working on this!

When we deprecate something, shouldn't we keep the old code path until the field is removed? Otherwise existing code could break.

File: cirq-google/cirq_google/calibration/workflow.py (hunk 1 above)
Comment (maffoo) [reply cirq-google/cirq_google/calibration/workflow.py:850]: I presume that's what happened; I didn't look into it too carefully.

**Comment by maffoo** [comment]
> When we deprecate something, shouldn't we keep the old code path until the field is removed? Otherwise existing code could break.

I did leave the parameters there, so code that passes `gate_set` explicitly will not break, but I did change things so values passed for `gate_set` are ignored. This seems ok to me because calls to the engine API will still work, it'll just use the circuit serializer instead of whatever was passed. Which serialization format is used feels to me like an internal implementation detail, so not something that needs to be preserved for a full deprecation cycle. But I'm of course happy to discuss other approaches.

**Review by verult — APPROVED** [review]
That makes sense, thanks for the explanation

**Pull request merged**